mitodl/mitxpro
# refactor: remove EdX-Api-Key usage

### What are the relevant tickets?
[#2872](https://github.com/mitodl/mitxpro/issues/2872)

### Description (What does it do?)
This PR removes EDX_API_KEY usage as Open edX is going to remove the EDX_API_KEY. Details can be found at https://github.com/openedx/edx-platform/issues/34039.

### Screenshots (if appropriate):


### How can this be tested?
- Fetch and check out to this branch, and check that edX is configured properly.
- Test all the endpoints that currently utilize `EdX-Api-Key` for client requests, or follow the testing instructions mentioned in https://github.com/mitodl/edx-api-client/pull/104. It's important to thoroughly test all the instructions provided in this [PR](https://github.com/mitodl/edx-api-client/pull/104).



### Additional Context
